### PR TITLE
docs: [Code Documentations] Update the smart contracts documentation

### PR DESCRIPTION
**Pull request by b00ste**
## What does this PR introduce?

- LSP6KeyManagerCore.sol - “Unknow Error” should be “Unknown Error”
- LSP5Utils.sol - "elemnt" should be "element"
- ILSP6KeyManager.sol - _calldata -> payload
- LSP2Utils.sol - Not every function has NatSpec.

### CODE REVIEW

File: contracts/LSP2ERC725YJSONSchema/LSP2Utils.sol [hunk 1]
```diff
@@ -16,10 +16,18 @@ library LSP2Utils {
 
     /* solhint-disable no-inline-assembly */
 
+    /**
+     * Generate a Singleton data key of `keyName`
+     * @param keyName The string that will be used to generate a Singleton data key
+     */
```
Comment (CJ42) [inline contracts/LSP2ERC725YJSONSchema/LSP2Utils.sol:22]: _Suggested changes_

What do you think @YamenMerhi, @skimaharvey?

```solidity
    /**
     * @dev Generates a data key of keyType Singleton
     * @param keyName The string to hash to generate a Singleton data key
     * @return a bytes32 dataKey
     * 
     */
```

File: contracts/LSP2ERC725YJSONSchema/LSP2Utils.sol [hunk 2]
```diff
@@ -179,6 +249,10 @@ library LSP2Utils {
         return true;
     }
 
+    /**
+     * Verifing if `data` is an encoded array of bytes4 (bytes4[])
```
Comment (CJ42) [inline contracts/LSP2ERC725YJSONSchema/LSP2Utils.sol:253]: Very important here to mention the term "ABI" to specify the encoding.

```suggestion
     * @dev verify that `data` is an array of bytes4 (bytes4[]) encoded according to the Solidity ABI specs.
```

File: contracts/LSP2ERC725YJSONSchema/LSP2Utils.sol [hunk 3]
```diff
@@ -124,6 +180,12 @@ library LSP2Utils {
         key = abi.encodePacked(bytes4(hashFunctionDigest), jsonDigest, url);
     }
 
+    /**
+     * Generate ASSET URL value
```
Comment (CJ42) [inline contracts/LSP2ERC725YJSONSchema/LSP2Utils.sol:184]:
```suggestion
     * @dev Generate a ASSETURL valueContent
```

File: contracts/LSP2ERC725YJSONSchema/LSP2Utils.sol [hunk 4]
```diff
@@ -104,6 +148,12 @@ library LSP2Utils {
         return bytes32(temporaryBytes);
     }
 
+    /**
+     * Generate a Mapping with Grouping data key by using a bytes12 and a bytes20
```
Comment (CJ42) [inline contracts/LSP2ERC725YJSONSchema/LSP2Utils.sol:152]:
```suggestion
     * @dev Generate a data key of keyType MappingWithGrouping
```

File: contracts/LSP2ERC725YJSONSchema/LSP2Utils.sol [hunk 5]
```diff
@@ -16,10 +16,18 @@ library LSP2Utils {
...
+    /**
+     * Generate a Singleton data key of `keyName`
+     * @param keyName The string that will be used to generate a Singleton data key
+     */
     function generateSingletonKey(string memory keyName) internal pure returns (bytes32) {
         return keccak256(bytes(keyName));
     }
 
+    /**
+     * Generating an Array data key for the `keyName`.
```
Comment (CJ42) [inline contracts/LSP2ERC725YJSONSchema/LSP2Utils.sol:28]:
```suggestion
     * @dev Generates a data key of keyType Array by hashing `keyName`.
```

File: contracts/LSP2ERC725YJSONSchema/LSP2Utils.sol [hunk 6]
```diff
@@ -44,6 +58,14 @@ library LSP2Utils {
         return bytes32(elementInArray);
     }
 
+    /**
+     * Generates a Mapping data key by using two strings:
```
Comment (CJ42) [inline contracts/LSP2ERC725YJSONSchema/LSP2Utils.sol:62]:
```suggestion
     * @dev Generates a data key of keyType Mapping by hashing two strings:
```

**Review by CJ42 — CHANGES_REQUESTED** [review]
I would suggest that we refer to keyType, valueContent, etc... from LSP2 to be more precise.

File: contracts/LSP2ERC725YJSONSchema/LSP2Utils.sol [hunk 7]
```diff
@@ -61,6 +83,13 @@ library LSP2Utils {
         return bytes32(temporaryBytes);
     }
 
+    /**
+     * Generates a Mapping data key by using one string and an address:
+     * <bytes10(keccak256(firstWord))>:<bytes20(addr)>
```
Comment (YamenMerhi) [inline contracts/LSP2ERC725YJSONSchema/LSP2Utils.sol:88]: Please point that there will be bytes2(0) at the middle in natspec

File: contracts/LSP2ERC725YJSONSchema/LSP2Utils.sol [hunk 8]
```diff
@@ -86,6 +121,15 @@ library LSP2Utils {
         return bytes32(generatedKey);
     }
 
+    /**
+     * Generate a Mapping with Grouping data key by using two strings and an address
+     * <bytes6(keccak256(firstWord))>:<bytes4(keccak256(secondWord))>:<bytes20(addr)>
```
Comment (YamenMerhi) [inline contracts/LSP2ERC725YJSONSchema/LSP2Utils.sol:126]: here also

**Review by YamenMerhi — CHANGES_REQUESTED** [review]

**Review by YamenMerhi — APPROVED** [review]

**Review by CJ42 — APPROVED** [review]

**Pull request merged**